modules/core/src/main/scala/org/tessellation/infrastructure/snapshot/GlobalSnapshotStorage.scala

```diff
@@ -60,6 +65,40 @@ object GlobalSnapshotStorage {
       }
   }
 
+  def make[F[_]: KryoSerializer](
+    ref: MapRef[F, SnapshotOrdinal, Option[GlobalSnapshot]]
+  )(implicit F: Async[F]): GlobalSnapshotStorage[F] =
+    new GlobalSnapshotStorage[F] {
+
+      implicit val ordinalOrder: Order[SnapshotOrdinal] = Order[Long].contramap[SnapshotOrdinal](_.value.value)
+      implicit val ordinalOrdering = ordinalOrder.toOrdering
+
+      def save(snapshot: GlobalSnapshot): F[Unit] =
+        getLast.flatMap { lastSnapshot =>
```

Comment:
This isn't correctly synchronized. It may lead to storing an invalid chain of snapshots. 

Notation used in example below to represent GlobalSnapshot: `ordinal: thisSnapshotHash -> lastSnapshotHash`

Initial state of the storage [`1: b -> a`]
Fiber 1 wants to save snapshot `2: c -> b`
Fiber 2 wants to save snapshot `2: x -> b`
Fiber 3 wants to save snapshot `3: d -> c`

Flow:
Fiber 1: Reads last snapshot, gets `1: b -> a`
Fiber 2: Reads last snapshot, gets `1: b -> a`
Fiber 1: Validates if the link `2: c -> b` follows `1: b -> a`, it does
Fiber 2: Validates if the link `2: x -> b` follows `1: b -> a`, it does
Fiber 1: Calls `ref(2).update(_ => (2: c -> b))`, storage contains now  [`2: c->b`, `1: b -> a`]
Fiber 3: Reads last snapshot, gets 
Fiber 3: Validates if the link `3: d -> c` follows `2: c -> b`, it does
Fiber 3: Calls `ref(3).update(_ => (3: d -> c))`, storage contains now  [`3: d -> c`, `2: c->b`, `1: b -> a`]
Fiber 2: Calls `ref(2).update(_ => (2: x -> b))`, storage contains now  [`3: d -> c`, `2: x->b`, `1: b -> a`]

The final chain is incorrect `[d, x, b]` while it should be `[d, c, b]`

Reply:
Ok, I started with another implementation. Now I can make it in a single modify as you did. I'll change that